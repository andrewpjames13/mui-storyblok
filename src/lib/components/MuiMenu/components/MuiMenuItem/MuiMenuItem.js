import React from 'react';
import PropTypes from 'prop-types';
import MenuItem from '@material-ui/core/MenuItem';
import Storyblok from '../../../../utils/Storyblok';
import appRedirect from '../../../../utils/appRedirect';

const MuiMenuItem = ({
  redirectRoute,
  buttonText,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  const redirect = () => appRedirect(redirectRoute);

  return (
    <MenuItem
      className={styles.root}
      onClick={redirect}
    >
      {buttonText}
    </MenuItem>
  );
};

export default MuiMenuItem;

MuiMenuItem.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** button text */
  buttonText: PropTypes.string.isRequired,
  /** redirect route */
  redirectRoute: PropTypes.string.isRequired,
};

MuiMenuItem.defaultProps = {
  rootClass: [],
};
