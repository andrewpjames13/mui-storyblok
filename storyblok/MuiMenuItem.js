module.exports = {
  name: 'MuiMenuItem',
  display_name: null,
  updated_at: '2020-04-01T16:44:24.342Z',
  schema: {
    'tab-aa85d3e7-f662-456c-8014-90ade7f002f1': {
      type: 'tab',
      display_name: 'route',
      keys: [],
      pos: 0,
    },
    buttonText: {
      type: 'text',
      required: true,
    },
    redirectRoute: {
      type: 'text',
      required: true,
    },
    legacyRedirect: {
      type: 'boolean',
    },
    googleConversionTracking: {
      type: 'option',
      use_uuid: true,
      options: [
        {
          value: 'contact',
          name: 'contact',
        },
        {
          value: 'prospect',
          name: 'prospect',
        },
      ],
    },
    rootClass: {
      type: 'options',
      options: [
        {
          value: 'color: white',
          name: 'color white',
        },
      ],
    },
  },
  image: null,
  preview_field: null,
  is_root: false,
  preview_tmpl: null,
  is_nestable: true,
  all_presets: [],
  preset_id: null,
  real_name: 'MuiMenuItem',
  component_group_uuid: null,
};
