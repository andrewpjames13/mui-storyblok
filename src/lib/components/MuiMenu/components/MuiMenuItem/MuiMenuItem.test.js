import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiMenuItem from './MuiMenuItem';

function setup(googleConversionTracking = '', legacyRedirect = true) {
  const props = {
    buttonText: 'buttonText',
    redirectRoute: 'page-welcome',
    legacyRedirect,
    googleConversionTracking,
  };
  const comp = mount(<MuiMenuItem {...props} />);
  return { comp, props };
}

describe('<MuiMenuItem />', () => {
  it('renders MuiMenuItem', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('handleClick and call GoogleHelpers contact ', async () => {
    const { comp } = setup('contact', false);
    window.location.assign = jest.fn();
    comp.find('ForwardRef(MenuItem)').at(0).simulate('click');
    expect(window.location.assign).toBeCalled();
  });

  test('snapshot with auth', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiMenuItem {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
