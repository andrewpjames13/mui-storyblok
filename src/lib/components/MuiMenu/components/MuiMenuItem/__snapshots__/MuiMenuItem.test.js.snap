// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiMenuItem /> snapshot with auth 1`] = `
<li
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-root-32 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
>
  buttonText
</li>
`;
